### PR TITLE
Data contribution guide

### PR DESCRIPTION
**Pull request by LaChapeliere**
# Description
Data contribution guide, define the format expected for contributions to the database + Scripts to upload and download the data.

Useful for issues #4 and #9 

## Type of change

Documentation + Feature

# How Has This Been Tested?
Tested on two csv files for each table (adapted from Fortune 21-30 PR).

# Checklist:

- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by LaChapeliere** [comment]
> @LaChapeliere Can I help with testing this somehow or do you need anything else?

I don't think we can test anything before we have the database up and running. Then we can write scripts to update it using the csv files, and that'll require a lot of testing! Any language you favour for that kind of backend tasks? I can do it in Python, you'll need to ask someone else if you've picked another language.

I did have a couple of questions though. Do you want a validation step of the uploaded content in the process (checking a few fields at random maybe to make sure people are not uploading fake data)? Do you want to keep all source CSVs on the repo or delete them once they've been uploaded to the database?

**Comment by kelseyhuse30** [comment]
> > @LaChapeliere Can I help with testing this somehow or do you need anything else?
> 
> I don't think we can test anything before we have the database up and running. Then we can write scripts to update it using the csv files, and that'll require a lot of testing! Any language you favour for that kind of backend tasks? I can do it in Python, you'll need to ask someone else if you've picked another language.
> 
> I did have a couple of questions though. Do you want a validation step of the uploaded content in the process (checking a few fields at random maybe to make sure people are not uploading fake data)? Do you want to keep all source CSVs on the repo or delete them once they've been uploaded to the database?

@LaChapeliere 

1. Python sounds great to me. 🐍
2. Validation sounds good but I think it can wait for now since we will review all data in PRs in Github before uploading.  Let's handle that once we get to a point where we think we need it.
3. I'm fine with either keeping or deleting - I would pick the one that takes less time.

**Comment by LaChapeliere** [comment]
I've finished the skeleton of the upload/download script, and the upload part for the basic company data.

@greendinosaur I've had to add a column and a UNIQUE contrainst manually to your scheme because I had forgotten to tell you that I had updated the diagram, but I don't think that will be a problem

@kelseyhuse30 You can start uploading the company data by using the script if you want. Your CSV need to have the exact format described in DATA_SOURCING.md. Drop any bug here so I can see if it's a code problem or a documentation problem and fix it :)

**Comment by LaChapeliere** [comment]
@kelseyhuse30 All ready for you to play with it!
I remain around for bugs and questions. It shouldn't be too hard to write queries to plug the database with your webpage, but if you're stuck ping me ;)

**Comment by kelseyhuse30** [comment]
@LaChapeliere thanks so much!! I may not have time to test this out until this weekend.  I will let you know how it goes. Really appreciate your work here ⭐ 💓

File: data/DATA_SOURCING.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+[A guide to CSV format](https://www.computerhope.com/issues/ch001356.htm)
+
+:exclamation: Please specify timestamps with the time zone, dates in YYYY-MM-DD format and numbers with dots as the decimal separator and no commas or unit symbols (e.g. 12000 and not $12.000).
+
+:exclamation: When a field contains commas, please [enclose it between quotes](https://stackoverflow.com/questions/4617935/is-there-a-way-to-include-commas-in-csv-columns-without-breaking-the-formatting) so it's not split by the csv reader!
+
+### Update the table with basic company information
+**TARGET = companies**
+
+This is the only table that accepts information on companies that are not in the database yet. Companies that already exist in the database will have their information updated (= overwritten), expect for fields where the new data is `null`.
```
Comment (kelseyhuse30) [inline data/DATA_SOURCING.md:48]:
```suggestion
This is the only table that accepts information on companies that are not in the database yet. Companies that already exist in the database will have their information updated (= overwritten), except for fields where the new data is `null`.
```

**Review by kelseyhuse30 — CHANGES_REQUESTED** [review]
This is awesome @LaChapeliere !! I was able to follow your instructions really easily.  I didn't upload new data, but I can see how it works and trust you have tested it.

I have one recommendation:
When I first ran the Python script, it prompted me to install the `pandas` package.  I think we should set up a python virtual env and freeze the dependencies into a `requirements.txt` in the `data` folder.  Then anyone who needs to install all dependencies can simply run `pip install -r requirements.txt`.  [This tutorial](https://packaging.python.org/tutorials/installing-packages/) explains how to do it.

Otherwise this looks great.  I can merge this now and add an Issue to set up the virtual env / requirements.txt file, or let you handle it in this PR.  Either are fine with me.  Do you have a preference?

Thank you again!!! 💓

**Comment by LaChapeliere** [comment]
@kelseyhuse30 My bad for the pandas import, I trashed the code that required pandas so I should have removed it. Since there's only one package to install now, the requirements.txt wouldn't make things any faster but I can still change to that setup if you prefer.

I do think it'd be better if someone tested the import, at least once for each table, mostly because I'd like to be sure that the doc is clear. But there's no hurry. If you prefer to merge and open issues tagging me later if things aren't working, that works for me too.

Glad I could help in any case! I don't think I'll get involved any further here because I already have too many projects that are closer to my skillsets, but I wish you'll get to see results for the work you're doing!

**Review by kelseyhuse30 — APPROVED** [review]

**Pull request merged**